nb-extension/js/widgets/DeclWidgetModel.js

```diff
@@ -1,46 +1,41 @@
 // Copyright (c) Jupyter Development Team.
 // Distributed under the terms of the Modified BSD License.
 
-define(function() {
+define(["nbextensions/widgets/widgets/js/widget"], function(Widget) {
     "use strict";
 
     /**
      * Collection of patches on top of ipywidgets.WidgetModel
      */
+    var DeclWidgetModel = Widget.WidgetModel.extend({
+        constructor: function() {
+            Widget.WidgetModel.apply(this, arguments);
+            // WidgetModel expects widgets' state to be set from kernel and won't
+            // set `_first_state` to false until that happens. But DeclWidgets
+            // are different: we do initial setup on client and then notify kernel.
+            this._first_state = false;
+        },
 
-    function register(WidgetManager, WidgetModel) {
-        var DeclWidgetModel = WidgetModel.extend({
-            constructor: function() {
-                WidgetModel.apply(this, arguments);
-                // WidgetModel expects widgets' state to be set from kernel and won't
-                // set `_first_state` to false until that happens. But DeclWidgets
-                // are different: we do initial setup on client and then notify kernel.
-                this._first_state = false;
-            },
-    
-            /*
-             * Making request_state a noop to avoid requiring this handshake to take place to
-             * create model.
-             */
-            request_state: function(callbacks) {
-                console.trace( "Empty implementation of request_state()");
-                return Promise.resolve(this);
-            },
-    
-            /*
-             * Avoiding out of sequence messages due to new promise/async code in WidgetModel
-             */
-            send_sync_message: function(attrs, callbacks) {
-                var data = {method: 'backbone', sync_data: attrs};
-                this.comm.send(data, callbacks);
-            }
-        });
-    
-        WidgetManager.register_widget_model('DeclWidgetModel', DeclWidgetModel);
-    }
+        /*
+         * Making request_state a noop to avoid requiring this handshake to take place to
+         * create model.
+         */
+        request_state: function(callbacks) {
+            console.trace( "Empty implementation of request_state()");
+            return Promise.resolve(this);
+        },
+
+        /*
+         * Avoiding out of sequence messages due to new promise/async code in WidgetModel
+         */
+        send_sync_message: function(attrs, callbacks) {
+            var data = {method: 'backbone', sync_data: attrs};
+            this.comm.send(data, callbacks);
+        }
+    });
 
     return {
-        register: register
+        DeclWidgetModel: DeclWidgetModel
```

Comment:
not a hack... seems to me that is what ipywidgets is doing too https://github.com/ipython/ipywidgets/blob/master/jupyter-js-widgets/src/widget_button.js#L90